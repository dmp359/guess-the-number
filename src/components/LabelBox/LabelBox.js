import React from 'react';
import ReactDOM from 'react-dom';
import LabelBox from './LabelBox';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LabelBox />, div);
});
