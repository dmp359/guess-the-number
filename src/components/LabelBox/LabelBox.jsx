import React from 'react';
import PropTypes from 'prop-types';

// Dummy component that displays a label and input box
const LabelBox = ({ label, onChange, value }) => {
  return (
    <div>
      <label>{label}</label>
      <input type="text" onChange={onChange} value={value} />
    </div>
  )
};
  
LabelBox.propTypes = {
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
};

export default LabelBox;
