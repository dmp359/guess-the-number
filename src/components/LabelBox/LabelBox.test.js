import React from 'react';
import { shallow } from 'enzyme';
import LabelBox from './LabelBox.jsx';

it('renders without crashing with all required and no optional value prop', () => {
  const component = shallow(
    <LabelBox label='label' onChange={() => {}} />
  );
  expect(component.debug()).toMatchSnapshot();
});

it('renders without crashing with all required and one optional value prop', () => {
  const component = shallow(
    <LabelBox label='label' onChange={() => {}} value='10' />
  );
  expect(component.debug()).toMatchSnapshot();
});


