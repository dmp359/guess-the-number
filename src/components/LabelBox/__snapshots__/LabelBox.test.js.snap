exports[`test renders without crashing with all required and no optional value prop 1`] = `
"<div>
  <label>
    label
  </label>
  <input type=\"text\" onChange={[Function: onChange]} value={[undefined]} />
</div>"
`;

exports[`test renders without crashing with all required and one optional value prop 1`] = `
"<div>
  <label>
    label
  </label>
  <input type=\"text\" onChange={[Function: onChange]} value=\"10\" />
</div>"
`;
